### PR TITLE
Automatic code fix suggestions from rubocop

### PR DESCRIPTION
**Pull request by mgrebenets**
merge! without block is same as overwrite
Use =~ instead of match
Use delete instead of g_sub
Use casecmp instead of downcase and ==

### CODE REVIEW

File: lib/fastlane/fast_file.rb [hunk 1]
```diff
@@ -105,7 +105,7 @@ def platform(platform_name, &block)
 
       self.current_platform = platform_name
 
-      block.call
+      yield
```
Comment (KrauseFx) [inline lib/fastlane/fast_file.rb]: Does that really work? I mean we have the block inside the `block` variable, right?

Comment (mgrebenets) [inline lib/fastlane/fast_file.rb]: Indeed, I took it off for now. What about the rest of the changes. Even with regards to https://github.com/fastlane/fastlane/pull/1155 these changes still look as some reasonable styling improvements and prepare the code for rubocop `0.36.0`.

**Comment by KrauseFx** [comment]
This can be closed now, right?

**Comment by mgrebenets** [comment]
Right, I'm tracking rubocop 0.36.0 related changes here: https://github.com/mgrebenets/fastlane/tree/rubocop-0.36.0

**Comment by KrauseFx** [comment]
Great :+1:

**Pull request closed without merging**